v0/destinations/google_cloud_function/transform.js

```diff
@@ -0,0 +1,154 @@
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  CustomError
+
+} = require("../../util");
+const {
+  TRIGGERTYPE
+} = require("./config");
+
+const {
+  generateBatchedPayloadForArray,
+  generateBatchedPayload
+} = require("./util");
+
+function process(event) {
+  const {
+      message,
+      destination
+  } = event;
+  const {
+      Config
+  } = destination;
+  const {
+      googleCloudFunctionUrl,
+      triggerType,
+      apiKeyId,
+      gcloudAuthorization
+  } = Config;
+  const basicAuth = Buffer.from(`apiKey:${apiKeyId}`).toString("base64");
+  if (!googleCloudFunctionUrl) {
+      throw new CustomError("Cloud Function Url not found. Aborting", 400);
```

Comment:
Also put a check for the scenario when https protocol is chosen and no gcloudAuthToken is not present